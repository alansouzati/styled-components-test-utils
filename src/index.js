export toBeAGlobalStyle from './matchers/toBeAGlobalStyle';
export toHaveComponent from './matchers/toHaveComponent';
export toHaveKeyframeRule from './matchers/toHaveKeyframeRule';
export toHaveStyleRule from './matchers/toHaveStyleRule';
export toNotHaveStyleRule from './matchers/toNotHaveStyleRule';
