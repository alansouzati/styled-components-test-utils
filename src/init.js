import styleSheet from 'styled-components/lib/models/StyleSheet';
import isOverV2 from './utils/isOverV2';
import isServer from './utils/isServer';

const init = () => {
  if (isOverV2()) {
    styleSheet.reset(isServer());
  }
};

export default init;
