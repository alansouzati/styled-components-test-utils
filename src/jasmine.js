import * as matchers from './';

export default function injectStyledUtils(jasmine) {
  const extension = {};
  Object.keys(matchers).forEach((x) => {
    extension[x] = () => ({
      compare: matchers[x],
    });
  });
  jasmine.addMatchers(extension);
}
