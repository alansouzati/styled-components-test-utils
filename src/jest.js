import * as matchers from './';
import toMatchStyledComponentsSnapshot from './matchers/toMatchStyledComponentsSnapshot';

const extension = {};
Object.keys(matchers).forEach((x) => {
  extension[x] = (...params) => matchers[x](...params);
});

extension.toMatchStyledComponentsSnapshot = toMatchStyledComponentsSnapshot;

expect.extend(extension);
