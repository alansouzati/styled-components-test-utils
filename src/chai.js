import init from './init';
// eslint-disable-next-line
import chai from 'chai';
import * as matchers from './';

init();

Object.keys(matchers).forEach((x) => {
  chai.Assertion.addMethod(x, function matcher(...params) {
    // eslint-disable-next-line
    const test = matchers[x](this._obj, ...params);

    this.assert(
      test.pass,
      test.message,
    );
  });
});
