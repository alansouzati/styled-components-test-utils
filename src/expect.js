import init from './init';
import * as matchers from './';

init();

export default function injectStyledUtils(expect) {
  const extension = {};
  Object.keys(matchers).forEach((x) => {
    extension[x] = function matcher(...params) {
      const test = matchers[x](this.actual, ...params);

      expect.assert(
        test.pass,
        test.message,
      );
      return this;
    };
  });

  expect.extend(extension);
}
