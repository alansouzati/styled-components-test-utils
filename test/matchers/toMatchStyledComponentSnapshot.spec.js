import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import styled from 'styled-components';
import { shallow, mount } from 'enzyme';
import '../../src/jest';

const Button = styled.section`
  padding: 4em;
  background: papayawhip;
`;

describe('toMatchStyledComponentsSnapshot', () => {
  test('null', () => {
    expect(null).toMatchStyledComponentsSnapshot();
  });

  test('should work with react-test-renderer', () => {
    const tree = ReactTestRenderer.create(
      <Button>Click here</Button>,
    ).toJSON();
    expect(tree).toMatchStyledComponentsSnapshot();
  });

  test('should work with enzyme shallow rendering', () => {
    const tree = shallow(<Button>Click here</Button>);
    expect(tree).toMatchStyledComponentsSnapshot();
  });

  test('should work with enzyme mount', () => {
    const tree = mount(<Button>Click here</Button>);
    expect(tree).toMatchStyledComponentsSnapshot();
  });
});
