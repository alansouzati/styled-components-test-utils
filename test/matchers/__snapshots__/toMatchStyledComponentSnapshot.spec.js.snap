// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toMatchStyledComponentsSnapshot null 1`] = `null`;

exports[`toMatchStyledComponentsSnapshot should work with enzyme mount 1`] = `
<styled.section>
  <section
    className="sc-bdVaJa iATjrR"
  >
    Click here
  </section>
</styled.section>
`;

exports[`toMatchStyledComponentsSnapshot should work with enzyme shallow rendering 1`] = `
<section
  className="sc-bdVaJa iATjrR"
>
  Click here
</section>
`;

exports[`toMatchStyledComponentsSnapshot should work with react-test-renderer 1`] = `
<section
  className="sc-bdVaJa iATjrR"
>
  Click here
</section>
`;
